torch/csrc/jit/script/module.h

```diff
@@ -361,15 +363,22 @@ struct NamedModule {
 struct NamedIValue {
   NamedIValue(std::string name, TypePtr type, IValue ivalue)
       : name_(name),
-        type(type),
-        ivalue(torch::make_unique<IValue>(std::move(ivalue))) {}
+        type_(type),
+        ivalue_(torch::make_unique<IValue>(std::move(ivalue))) {}
 
-  IValue* slot() const {
-    return ivalue.get();
+  Slot slot() const {
+    return Slot(ivalue_.get());
```

Comment:
Oh, I see now, the lifetimes of `IValue` and `Slot` are  the same unless someone creates the `Slot` outside of a `NamedValue`. Would it make sense to make `Slot` c-tor private, so only `NamedValue`s can create `Slot`s?